library/alloc/src/string.rs

```diff
@@ -2444,6 +2452,49 @@ impl ToString for i8 {
     }
 }
 
+#[cfg(not(no_global_oom_handling))]
+#[stable(feature = "u16_to_string_specialization", since = "0.00.0")]
+impl ToString for u16 {
+    #[inline]
+    fn to_string(&self) -> String {
+        let mut buf = String::with_capacity(u16::MAX.log10() as usize + 1);
+        fmt_with_buf(&mut buf, self);
+        buf
+    }
+}
+
+#[cfg(not(no_global_oom_handling))]
+#[stable(feature = "i16_to_string_specialization", since = "0.00.0")]
+impl ToString for i16 {
+    #[inline]
+    fn to_string(&self) -> String {
+        let mut buf = String::with_capacity(
+            i16::MAX.log10() as usize + 1 /* and incorporate the possible minus sign: */ + 1,
+        );
```

Comment:
If I know answer is 6, and you know the answer is 6, why make the compiler do all this extra work to figure it out?
Also unless you explicitly store the result in a const variable, it isn't guaranteed to be evaluated at compile time afaik (at least, it isn't in an unoptimised build https://godbolt.org/z/eYGsr9bYj).

Reply:
> If I know answer is 6, and you know the answer is 6, why make the compiler do all this extra work to figure it out?

Because it's less of a magic number then.

> Also unless you explicitly store the result in a const variable, it isn't guaranteed to be evaluated at compile time afaik (at least, it isn't in an unoptimised build https://godbolt.org/z/eYGsr9bYj).

With `opt-level=1` it is but sure I could put it in a `const` to make sure it's always inlined.